code/hash/n-merge-bang-vs-[]=.rb

```diff
@@ -0,0 +1,22 @@
+require 'benchmark/ips'
+
+ENUM = (1..100)
+HASH = ENUM.to_h{ |a| [a, a]}
+SECOND_HASH = ENUM.to_h{ |a| [a+50, a+50]}
+
+def merge
+  h = HASH
+  h.merge!(SECOND_HASH)
```

Comment:
This changes the (not longer) constant `HASH` and that seem to be intended. You can just use `HASH#merge`.

Reply:
`[]=` also changes `HASH`

Reply:
@ParadoxV5 ?!
Do you mean the assignment to `h` below? In this case `h` is created in line 13…